test/conformance/event/urEventGetInfo.cpp

```diff
@@ -1,2 +1,137 @@
+        }
+        uur::urQueueTestWithParam<T>::TearDown();
+    }
+
+    const size_t count = 1024;
+    const size_t size = sizeof(uint32_t) * count;
+    ur_mem_handle_t buffer = nullptr;
+    ur_event_handle_t event = nullptr;
+    std::vector<uint32_t> input;
+};
+
+using urEventGetInfoTest = urEventTestWithParam<ur_event_info_t>;
+
+TEST_P(urEventGetInfoTest, Success) {
+
+    ur_event_info_t info_type = getParam();
+    size_t size;
+    ASSERT_SUCCESS(urEventGetInfo(event, info_type, 0, nullptr, &size));
+    ASSERT_NE(size, 0);
+    std::vector<uint8_t> data(size);
+    ASSERT_SUCCESS(
+        urEventGetInfo(event, info_type, size, data.data(), nullptr));
+
+    switch (info_type) {
+    case UR_EVENT_INFO_COMMAND_QUEUE: {
+        auto returned_queue = reinterpret_cast<ur_queue_handle_t>(data.data());
+        EXPECT_EQ(queue, returned_queue);
+        break;
+    }
+    case UR_EVENT_INFO_CONTEXT: {
+        auto returned_context =
+            reinterpret_cast<ur_context_handle_t>(data.data());
+        EXPECT_EQ(context, returned_context);
+        break;
+    }
+    case UR_EVENT_INFO_COMMAND_TYPE: {
+        auto returned_command = reinterpret_cast<ur_command_t *>(data.data());
+        EXPECT_EQ(UR_COMMAND_MEM_BUFFER_WRITE, *returned_command);
+        break;
+    }
+    case UR_EVENT_INFO_COMMAND_EXECUTION_STATUS: {
+        auto returned_status =
+            reinterpret_cast<ur_event_status_t *>(data.data());
+        EXPECT_EQ(UR_EVENT_STATUS_COMPLETE, *returned_status);
+        break;
+    }
+    case UR_EVENT_INFO_REFERENCE_COUNT: {
+        auto returned_reference_count =
+            reinterpret_cast<uint32_t *>(data.data());
+        EXPECT_EQ(1, *returned_reference_count);
+        break;
+    }
+    default:
+        ASSERT_FALSE(false);
+    }
+}
+
+UUR_TEST_SUITE_P(urEventGetInfoTest,
+                 ::testing::Values(UR_EVENT_INFO_COMMAND_QUEUE,
+                                   UR_EVENT_INFO_CONTEXT,
+                                   UR_EVENT_INFO_COMMAND_TYPE,
+                                   UR_EVENT_INFO_COMMAND_EXECUTION_STATUS,
+                                   UR_EVENT_INFO_REFERENCE_COUNT),
+                 uur::deviceTestWithParamPrinter<ur_event_info_t>);
+
+using urEventGetInfoNegativeTest = uur::event::urEventTest;
+
+TEST_P(urEventGetInfoNegativeTest, InvalidNullHandle) {
+    ur_event_info_t info_type = UR_EVENT_INFO_COMMAND_QUEUE;
+    size_t size;
+    ASSERT_SUCCESS(urEventGetInfo(event, info_type, 0, nullptr, &size));
+    ASSERT_NE(size, 0);
+    std::vector<uint8_t> data(size);
+
+    /* Invalid hEvent */
+    EXPECT_EQ(
+        urEventGetInfo(nullptr, UR_EVENT_INFO_COMMAND_QUEUE, 0, nullptr, &size),
+        UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+
+    /* Valid hEvent but invalid pPropValue and pPropValueSizeRet */
+    EXPECT_EQ(
+        urEventGetInfo(event, UR_EVENT_INFO_COMMAND_QUEUE, 0, nullptr, nullptr),
+        UR_RESULT_ERROR_INVALID_NULL_HANDLE);
```

Comment:
I think this is a valid invocation - should return `UR_RESULT_SUCCESS`. Essentially does nothing though.

Reply:
I think you can just remove it.

Reply:
On second thought, both parameters are marked as optional. So I guess this behaviour is implied. I don't mind removing it then